I have some compile errors with my change to wrap the APIs with "experimental" namespace as was suggested by @aliuTT, don't know what is up there (didn't see it locally), will look closer.

https://github.com/tenstorrent/tt-metal/actions/runs/13020087172/job/36318876275

```
/home/ubuntu/actions-runner/_work/tt-metal/tt-metal/ttnn/cpp/ttnn/operations/ccl/ccl_host_datastructures.hpp:18:55: error: reference to 'experimental' is ambiguous
   18 |     std::size_t total_l1_buffer_space = tt::tt_metal::experimental::hal::get_erisc_l1_unreserved_size();
      |                                         ~~~~~~~~~~~~~~^
/home/ubuntu/actions-runner/_work/tt-metal/tt-metal/tt_metal/api/tt-metalium/hal_exp.hpp:10:25: note: candidate found by name lookup is 'tt::tt_metal::experimental'
   10 | namespace tt::tt_metal::experimental::hal {
      |                         ^
/home/ubuntu/actions-runner/_work/tt-metal/tt-metal/tt_metal/api/tt-metalium/host_api.hpp:887:11: note: candidate found by name lookup is 'tt::tt_metal::v0::experimental'
  887 | namespace experimental {
      |           ^
```

--- a/docs/source/tt-metalium/tt_metal/apis/host_apis/command_queue/LightMetalBeginCapture.rst
+++ b/docs/source/tt-metalium/tt_metal/apis/host_apis/command_queue/LightMetalBeginCapture.rst
@@ -1,0 +1,4 @@
+LightMetalBeginCapture
+======================
+
+.. doxygenfunction:: tt::tt_metal::v0::LightMetalBeginCapture

--- a/docs/source/tt-metalium/tt_metal/apis/host_apis/command_queue/LightMetalEndCapture.rst
+++ b/docs/source/tt-metalium/tt_metal/apis/host_apis/command_queue/LightMetalEndCapture.rst
@@ -1,0 +1,4 @@
+LightMetalEndCapture
+====================
+
+.. doxygenfunction:: tt::tt_metal::v0::LightMetalEndCapture

--- a/docs/source/tt-metalium/tt_metal/apis/host_apis/command_queue/command_queue.rst
+++ b/docs/source/tt-metalium/tt_metal/apis/host_apis/command_queue/command_queue.rst
@@ -16,5 +16,7 @@ CommandQueue
   ReplayTrace
   ReleaseTrace
   EnqueueTrace
+  LightMetalBeginCapture
+  LightMetalEndCapture
   Finish
   Synchronize

--- a/tt_metal/api/tt-metalium/host_api.hpp
+++ b/tt_metal/api/tt-metalium/host_api.hpp
@@ -12,6 +12,7 @@
 #include "device.hpp"
 #include "sub_device_types.hpp"
 #include "span.hpp"
+#include "lightmetal_binary.hpp"
 
 /** @file */
 
@@ -882,6 +883,27 @@ void ReleaseTrace(IDevice* device, const uint32_t tid);
  */
 // clang-format on
 void EnqueueTrace(CommandQueue& cq, uint32_t trace_id, bool blocking);
+
+// clang-format off
+/**
+ * Begin Light Metal Binary capturing on host and all devices. This will trace host API calls and device (metal trace) workloads to a
+ * binary blob returned to caller when tracing is finished, which can later be rerun directly from binary.
+ * Note: This LightMetalBinary Trace/Replay feature is currently under active development and is not fully supported, use at own risk.
+ *
+ * Return value: void
+ */
+// clang-format on
+void LightMetalBeginCapture();
+
+// clang-format off
+/**
+ * Ends Light Metal Binary capturing on host and all devices returns the binary blob to the user.
+ * Note: This LightMetalBinary Trace/Replay feature is currently under active development and is not fully supported, use at own risk.
+ *
+ * Return value: LightMetalBinary
+ */
+// clang-format on
+LightMetalBinary LightMetalEndCapture();
 
 // clang-format off
 /**

--- a/tt_metal/api/tt-metalium/lightmetal_binary.hpp
+++ b/tt_metal/api/tt-metalium/lightmetal_binary.hpp
@@ -1,0 +1,64 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <vector>
+#include <cstdint>
+#include <string>
+#include <fstream>
+#include <stdexcept>
+#include <iostream>
+
+namespace tt::tt_metal {
+
+// Public interface wrapper for serialized LightMetalBinary data
+class LightMetalBinary {
+private:
+    std::vector<uint8_t> data_;
+
+public:
+    // Defalt constructor, and constructor from raw data vector.
+    LightMetalBinary() = default;
+    explicit LightMetalBinary(std::vector<uint8_t> data) : data_(std::move(data)) {}
+
+    // Public accesors for the binary data
+    const std::vector<uint8_t>& get_data() const { return data_; }
+    void set_data(std::vector<uint8_t> data) { data_ = std::move(data); }
+    size_t size() const { return data_.size(); }
+    bool is_empty() const { return data_.empty(); }
+
+    // Save binary data to a file
+    void save_to_file(const std::string& filename) const {
+        std::ofstream outFile(filename, std::ios::binary);
+        if (!outFile.is_open()) {
+            throw std::ios_base::failure("Failed to open file: " + filename);
+        }
+        outFile.write(reinterpret_cast<const char*>(data_.data()), data_.size());
+        if (!outFile.good()) {
+            throw std::ios_base::failure("Failed to write to file: " + filename);
+        }
+    }
+
+    // Load binary data from a file
+    static LightMetalBinary load_from_file(const std::string& filename) {
+        std::ifstream in_file(filename, std::ios::binary | std::ios::ate);
+        if (!in_file.is_open()) {
+            throw std::ios_base::failure("Failed to open file: " + filename);
+        }
+        auto file_size = in_file.tellg();
+        if (file_size <= 0) {
+            throw std::runtime_error("File is empty or error occurred while reading: " + filename);
+        }
+        std::vector<uint8_t> buffer(file_size);
+        in_file.seekg(0, std::ios::beg);
+        in_file.read(reinterpret_cast<char*>(buffer.data()), file_size);
+        if (!in_file.good()) {
+            throw std::ios_base::failure("Failed to read file: " + filename);
+        }
+        return LightMetalBinary(std::move(buffer));
+    }
+};
+
+}  // namespace tt::tt_metal

--- a/tt_metal/tt_metal.cpp
+++ b/tt_metal/tt_metal.cpp
@@ -1364,6 +1364,16 @@ void ReplayTrace(IDevice* device, const uint8_t cq_id, const uint32_t tid, const
 
 void ReleaseTrace(IDevice* device, const uint32_t tid) { device->release_trace(tid); }
 
+// Light Metal Begin/End Capture APIs are stubs for now, filled in soon.
+void LightMetalBeginCapture() {
+    log_warning(tt::LogMetalTrace, "Begin LightMetalBinary Capture - not yet implemented.");
+}
+
+LightMetalBinary LightMetalEndCapture() {
+    log_warning(tt::LogMetalTrace, "End LightMetalBinary Capture - not yet implemented.");
+    return {};
+}
+
 void Synchronize(IDevice* device, const std::optional<uint8_t> cq_id, tt::stl::Span<const SubDeviceId> sub_device_ids) {
     if (std::getenv("TT_METAL_SLOW_DISPATCH_MODE") == nullptr) {
         if (cq_id.has_value()) {

